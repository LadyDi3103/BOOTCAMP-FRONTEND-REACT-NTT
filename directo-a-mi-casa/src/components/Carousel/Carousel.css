.carousel {
    position: relative;
    width: 100%;
    height: auto;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: var(--color-surface);
}

.carousel img {
    width: 100%;
    height: auto;
    object-fit: cover;
}

.carousel-mobile {
    display: block;
}

.carousel-desktop {
    display: none;
}

.carousel img {
    width: 100%;
    height: auto;
    object-fit: cover;
}

@media (min-width: 768px) and (max-width: 1023px) {
    .carousel-mobile {
        display: none;
    }

    .carousel-desktop {
        display: block;
    }
}

@media (min-width: 1024px) {
    .carousel-mobile {
        display: none;
    }

    .carousel-desktop {
        display: block;
    }
}

@media (min-width: 1300px) {

    .carousel-desktop {
        max-height: 600px;
    }
}