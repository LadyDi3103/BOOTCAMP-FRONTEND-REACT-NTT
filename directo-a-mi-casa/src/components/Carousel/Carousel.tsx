import React from 'react';

/**
 * Componente Carousel
 * Muestra im치genes en formato de carrusel para dispositivos m칩viles y de escritorio.
 */
const Carousel: React.FC = () => {
    return (
        <section className="carousel">
            {/* Imagen para dispositivos m칩viles */}
            <img
                src="/src/assets/images/carousel/HP-BANNER-PRINCIPAL-mobile.svg"
                alt="Imagen para dispositivos m칩viles"
                className="carousel-img carousel-mobile"
            />

            {/* Imagen para dispositivos de escritorio */}
            <img
                src="/src/assets/images/carousel/baner_desktop.webp"
                alt="Imagen para dispositivos de escritorio"
                className="carousel-img carousel-desktop"
            />
        </section>
    );
};

export default Carousel;
