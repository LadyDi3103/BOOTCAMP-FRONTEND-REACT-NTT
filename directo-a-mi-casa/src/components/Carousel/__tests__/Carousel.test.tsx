import { render, screen } from "@testing-library/react";
import Carousel from "../Carousel";

describe("Carousel Component", () => {
    test("should render the mobile image correctly", () => {
        render(<Carousel />);

        const mobileImage = screen.getByAltText("Imagen para dispositivos mÃ³viles");
        expect(mobileImage).toBeInTheDocument();
        expect(mobileImage).toHaveAttribute("src", "/src/assets/images/carousel/HP-BANNER-PRINCIPAL-mobile.svg");
    });

    test("should render the desktop image correctly", () => {
        render(<Carousel />);

        const desktopImage = screen.getByAltText("Imagen para dispositivos de escritorio");
        expect(desktopImage).toBeInTheDocument();
        expect(desktopImage).toHaveAttribute("src", "/src/assets/images/carousel/baner_desktop.webp");
    });

    test("should show broken image or missing alt attribute", () => {
        render(
            <section className="carousel">
                <img
                    src="/src/assets/images/carousel/non-existent-image.svg"
                    alt=""
                    className="carousel-img carousel-mobile"
                />
            </section>
        );

        const brokenImage = screen.getByAltText("");
        expect(brokenImage).toHaveAttribute("src", "/src/assets/images/carousel/non-existent-image.svg");
    });

    test("Should match snapshot", () => {
        const { container } = render(<Carousel />);
        expect(container).toMatchSnapshot();
    });
});


