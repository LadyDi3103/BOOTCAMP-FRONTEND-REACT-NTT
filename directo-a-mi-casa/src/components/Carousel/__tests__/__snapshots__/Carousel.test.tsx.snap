// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component Should match snapshot 1`] = `
<div>
  <section
    class="carousel"
  >
    <img
      alt="Imagen para dispositivos mÃ³viles"
      class="carousel-img carousel-mobile"
      src="/src/assets/images/carousel/HP-BANNER-PRINCIPAL-mobile.svg"
    />
    <img
      alt="Imagen para dispositivos de escritorio"
      class="carousel-img carousel-desktop"
      src="/src/assets/images/carousel/baner_desktop.webp"
    />
  </section>
</div>
`;
